pkg/operation/botanist/botanist.go

```diff
@@ -163,7 +168,13 @@ func (b *Botanist) UnregisterAsSeed() error {
 	if err := b.K8sGardenClient.Garden().GardenV1beta1().Seeds().Delete(seed.Name, nil); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-	if err := b.K8sGardenClient.DeleteSecret(seed.Spec.SecretRef.Namespace, seed.Spec.SecretRef.Name); err != nil && !apierrors.IsNotFound(err) {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      seed.Spec.SecretRef.Name,
+			Namespace: seed.Spec.SecretRef.Namespace,
+		},
+	}
+	if err := b.K8sGardenClient.Client().Delete(context.TODO(), secret, kubernetes.DefaultDeleteOptionFuncs...); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
After #1106 is merged you can use `client.IgnoreNotFound`

I probably missed a lot of places, can you check yourself? :)